run_release.py

```diff
@@ -316,6 +316,23 @@ def check_ssh_connection(db: ReleaseShelf) -> None:
     client.exec_command("pwd")
 
 
+def check_sigstore_client(db: ReleaseShelf) -> None:
+    client = paramiko.SSHClient()
+    client.load_system_host_keys()
+    client.set_missing_host_key_policy(paramiko.WarningPolicy)
+    client.connect(DOWNLOADS_SERVER, port=22, username=db["ssh_user"])
+    _, stdout, _ = client.exec_command("python3 -m sigstore --version")
+    sigstore_version = stdout.read(1000).decode()
+    sigstore_vermatch = re.match("^sigstore ([0-9.]+)")
```

Comment:
Looks like type checking is useful after all ;)

```python
mypy: commands[0]> mypy .
run_release.py:326: error: No overload variant of "match" matches argument type
"str"  [call-overload]
        sigstore_vermatch = re.match("^sigstore ([0-9.]+)")
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
run_release.py:326: note: Possible overload variants:
run_release.py:326: note:     def match(pattern: str | Pattern[str], string: str, flags: int | RegexFlag = ...) -> Match[str] | None
run_release.py:326: note:     def match(pattern: bytes | Pattern[bytes], string: Buffer, flags: int | RegexFlag = ...) -> Match[bytes] | None
Found 1 error in 1 file (checked 13 source files)
mypy: exit 1 (7.00 seconds) /home/runner/work/release-tools/release-tools> mypy . pid=[20](https://github.com/python/release-tools/actions/runs/11936459002/job/33270152025?pr=194#step:6:21)29
```

https://github.com/python/release-tools/actions/runs/11936459002/job/33270152025?pr=194


```pytb
💥  Checking Sigstore CLI
Traceback (most recent call last):
  File "/Users/hugo/github/release-tools/run_release.py", line 1336, in <module>
    main()
  File "/Users/hugo/github/release-tools/run_release.py", line 1332, in main
    automata.run()
  File "/Users/hugo/github/release-tools/run_release.py", line 245, in run
    raise e from None
  File "/Users/hugo/github/release-tools/run_release.py", line 242, in run
    self.current_task(self.db)
  File "/Users/hugo/github/release-tools/release.py", line 126, in __call__
    return getattr(self, "function")(db)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hugo/github/release-tools/run_release.py", line 326, in check_sigstore_client
    sigstore_vermatch = re.match("^sigstore ([0-9.]+)")
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: match() missing 1 required positional argument: 'string'
```

Reply:
Hahah, true. Let me fix that!